service/lib/dinstaller/dbus/software.rb

```diff
@@ -64,7 +69,32 @@ def initialize(backend, logger)
         dbus_method :ProvisionsSelected, "in Provisions:as, out Result:ab" do |provisions|
           [provisions.map { |p| backend.provision_selected?(p) }]
         end
+
+        dbus_method :Propose do
+          backend.propose(progress_handler)
+        end
+
+        dbus_method :Probe do
+          backend.probe(progress_handler)
+        end
+
+        dbus_method :Install do
+          backend.install(progress_handler)
+        end
+
+        dbus_method :Finish do
+          backend.finish(progress_handler)
+        end
+
+        # Progress has struct with values:
+        #   s message
+        #   t total major steps to do
+        #   t current major step (0-based)
+        #   t total minor steps. Can be zero which means no minor steps
+        #   t current minor step
+        dbus_reader :progress, "(stttt)"
```

Comment:
Maybe we should start ignoring the minor steps. I think we will drop minor steps sooner than later.

Reply:
I agree. But I preferred to keep them consistent until we have a final working version. Perhaps we find those bits useful at the end.